### PR TITLE
Support new tarfile types from parser

### PR DESCRIPTION
**Pull request by lozette**
## Changes in this PR:

Support an alternative tarfile structure.

TODO: the final metadata format isn't decided but this tries to cope with an unknown structure as best as it can, until we have the metadata nailed down.

## Trello card / Rollbar error (etc)

https://trello.com/c/BnOQ0JuX/1033-%F0%9F%97%BBendpoint-for-new-documents

and 

https://trello.com/c/bAB9Al9P/1038-%F0%9F%97%BBwhat-info-goes-in-the-editor-notification-when-a-judgment-doesnt-come-from-tdr

and

https://rollbar.com/dxw/tna-caselaw-ingester/items/45/
https://rollbar.com/dxw/tna-caselaw-ingester/items/44/

- [ ] Requires env variable(s) to be updated

### CODE REVIEW

File: ds-caselaw-ingester/lambda_function.py [hunk 1]
```diff
@@ -116,7 +116,10 @@ def extract_lambda_versions(versions: List[Dict[str, str]]) -> List[Tuple[str, s
 
 
 def store_metadata(uri: str, metadata: dict) -> None:
-    tdr_metadata = metadata["parameters"]["TDR"]
+    try:
+        tdr_metadata = metadata["parameters"]["TDR"]
+    except KeyError:
+        return  # TODO: Get non-TDR metadata added to the metadata file
```
Comment (dragon-dxw) [inline ds-caselaw-ingester/lambda_function.py:122]: Would a printed error message make sense here?

File: ds-caselaw-ingester/lambda_function.py [hunk 2]
```diff
@@ -354,7 +378,7 @@ def handler(event, context):
     docx_filename = extract_docx_filename(metadata, consignment_reference)
     copy_file(
         tar,
-        f"{consignment_reference}/{docx_filename}",
+        f"{docx_filename}",
```
Comment (dragon-dxw) [inline ds-caselaw-ingester/lambda_function.py:381]: This feels like it'd be a breaking change -- I can't find where we're changing the existing behaviour to embed the consignment_reference in the docx or move the file so it's not in a consignment_reference folder.

Comment (lozette) [reply ds-caselaw-ingester/lambda_function.py]: Amended!

File: README.md [hunk 3]
```diff
@@ -80,6 +80,18 @@ And then send a message:
...
+## Local testing
+
+To test a tarfile locally:
+
+1. Add your test tarfile to `aws_examples/s3/te-editorial-out-int`
+2. Edit `aws_examples/sns/parsed-judgment.json` to contain your tarfile name in `s3-folder-url` and consignment reference
+   in `consignment-reference`
+3. In `scripts/setup-localstack.sh`, change the line which begins `awslocal s3 cp aws_examples/s3/te-editorial-out-int/`
+   to contain your tarfile name, e.g. if your tarfile is called `XYZ-123.tar.gz`, this line should read
+   `awslocal s3 cp aws_examples/s3/te-editorial-out-int/XYZ-123.tar.gz s3://te-editorial-out-int`
```
[the remark this answers is not in the document]
Comment (lozette) [reply README.md:92]: `setup-localstack.sh` is called from `make setup`, so we'd need to pass a param from `make setup` to `setup-localstack.sh`, and yes it looks super finicky!

[the remark this answers is not in the document]
Comment (lozette) [reply README.md]: Added in a new commit!

File: ds-caselaw-ingester/lambda_function.py (hunk 1 above)
Comment (lozette) [reply ds-caselaw-ingester/lambda_function.py:122]: I'm not going to commit this until we have something else for the metadata, the TODO is to remind me to add it!

File: ds-caselaw-ingester/lambda_function.py [hunk 4]
```diff
@@ -116,27 +116,45 @@ def extract_lambda_versions(versions: List[Dict[str, str]]) -> List[Tuple[str, s
...
+            uri,
+            name="transfer-consignment-reference",
+            value=tdr_metadata["Internal-Sender-Identifier"],
+        )
+        api_client.set_property(
+            uri,
+            name="transfer-received-at",
+            value=tdr_metadata["Consignment-Completed-Datetime"],
+        )
+    except KeyError:
```
Comment (dragon-dxw) [inline ds-caselaw-ingester/lambda_function.py:142]: Can we be more explicit about this? Perhaps we switch if a specific field does/does not exist (e.g. if there's not a `TDR` field, or a `fakeTDR` exists?)

Alternatively,

```
try:
        tdr_metadata = metadata["parameters"]["TDR"]
except KeyError:
       ...
else:
      normal_route
```
could narrow the failing line down to one?

**Comment by jacksonj04** [comment]
Closing this as stale. If it's important, it'll come back.

**Pull request closed without merging**